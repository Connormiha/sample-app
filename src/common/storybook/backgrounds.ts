import 'common/storybook/background.pcss';

export const backgrounds = {
    backgrounds: [
        {name: 'light-gray', value: '#f5f6fa', default: true},
        {name: 'White', value: '#fff'},
        {name: 'Black', value: '#000'},
        {name: 'Gray', value: '#ccc'},
        {name: 'Twitter', value: '#00aced'},
        {name: 'Facebook', value: '#3b5998'},
        {name: 'Ok.ru', value: '#ee7808'}
    ]
};
