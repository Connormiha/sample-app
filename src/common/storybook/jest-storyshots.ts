import initStoryshots from '@storybook/addon-storyshots';

export const requireSnapshot = (path: string): void => {
    const config: any = {
        config: ({configure}: any): any => {
            configure(() => {
                require(path.replace('.test', '.story'));
            }, module);
        },
        framework: 'react'
    };

    initStoryshots(config);
};
