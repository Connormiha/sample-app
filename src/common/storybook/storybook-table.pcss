.storybook-table {
    padding: 0 20px 20px;

    &__title {
        font-size: 15px;
    }

    &__table {
        border-collapse: collapse;
    }

    &__td {
        border: 0;
        padding: 0 20px 0 0;
        vertical-align: top;

        &_width {
            &_small {
                width: 100px;
            }

            &_medium {
                width: 170px;
            }

            &_large {
                width: 370px;
            }
        }
    }
}
