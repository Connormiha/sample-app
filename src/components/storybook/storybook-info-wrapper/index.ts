export * from './storybook-info-wrapper';
export {default} from './storybook-info-wrapper';
