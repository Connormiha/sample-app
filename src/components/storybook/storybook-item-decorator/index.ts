export * from './storybook-item-decorator';
export {default} from './storybook-item-decorator';
