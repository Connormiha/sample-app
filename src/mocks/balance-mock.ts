import {IBalanceInfo} from 'types/api';

export const balanceResult: IBalanceInfo = {
    balance: 213920,
    next_payout: 159465,
    currency: 'rub'
};
