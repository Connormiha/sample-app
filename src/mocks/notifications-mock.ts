import {INotification} from 'stores/notification-item-store';

export const notifications: INotification[] = [
    {
        type: 'error',
        title: 'Wrong User id',
        message: 'There is no such user with id 100500'
    },
    {
        type: 'warning',
        title: 'Not all required fields ready',
        message: 'There are name warnings'
    },
    {
        type: 'info',
        title: 'Creating user complete',
        message: 'User with name John was created'
    }
];
