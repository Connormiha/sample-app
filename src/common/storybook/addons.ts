import 'common/css/global.pcss';
import 'common/css/ui-variables.pcss';
import 'common/css/ui-animations.pcss';
import 'common/storybook/global.pcss';
import {addParameters, setAddon} from '@storybook/react';
import {backgrounds} from './backgrounds';
import JSXAddon from 'storybook-addon-jsx';

addParameters(backgrounds);
setAddon(JSXAddon);
