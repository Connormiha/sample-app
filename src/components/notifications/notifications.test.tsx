import {requireSnapshot} from 'common/storybook/jest-storyshots';
import * as React from 'react';
import {mount} from 'enzyme';
import Notifications from './notifications';
import NotificationsStore from 'stores/notifications-store';
import {Provider} from 'mobx-react';
import {notifications} from 'mocks/notifications-mock';

requireSnapshot(__filename);

describe('<Notifications />', () => {
    it('should close notification', () => {
        const store = new NotificationsStore();
        notifications.forEach((item) => store.add(item));

        const component = mount(
            <Provider notifications={store}>
                <Notifications />
            </Provider>
        );

        const closeButtons = component.find('.notification-item__close');
        expect(closeButtons).toHaveLength(notifications.length);
        closeButtons.at(0).simulate('click');
        closeButtons.at(0).simulate('animationend');
        expect(component.find('.notification-item__close')).toHaveLength(notifications.length - 1);
    });
});
