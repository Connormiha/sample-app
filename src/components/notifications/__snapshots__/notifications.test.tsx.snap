// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Notifications Default 1`] = `
<div
  className="notifications"
>
  <div
    className="notification-item notification-item_type_error"
    onAnimationEnd={[Function]}
  >
    <div
      className="notification-item__content"
    >
      <div
        className="notification-item__title"
      >
        Wrong User id
      </div>
      <div
        className="notification-item__message"
      >
        There is no such user with id 100500
      </div>
    </div>
    <button
      className="notification-item__close"
      onClick={[Function]}
      title="Закрыть"
    >
      X
    </button>
  </div>
  <div
    className="notification-item notification-item_type_warning"
    onAnimationEnd={[Function]}
  >
    <div
      className="notification-item__content"
    >
      <div
        className="notification-item__title"
      >
        Not all required fields ready
      </div>
      <div
        className="notification-item__message"
      >
        There are name warnings
      </div>
    </div>
    <button
      className="notification-item__close"
      onClick={[Function]}
      title="Закрыть"
    >
      X
    </button>
  </div>
  <div
    className="notification-item notification-item_type_info"
    onAnimationEnd={[Function]}
  >
    <div
      className="notification-item__content"
    >
      <div
        className="notification-item__title"
      >
        Creating user complete
      </div>
      <div
        className="notification-item__message"
      >
        User with name John was created
      </div>
    </div>
    <button
      className="notification-item__close"
      onClick={[Function]}
      title="Закрыть"
    >
      X
    </button>
  </div>
</div>
`;
